### PR TITLE
Implement basic integration test infra for BESS

### PR DESCRIPTION
**Pull request by osinstom**
This PR enables running integration tests for BESS. Also, I move all Go code related to test infra to `framework.go`, so that `basic_test.go` contains test cases only. 

Note that the BESS forwarding state is not verified yet by integration tests.

### CODE REVIEW

File: test/integration/framework.go [hunk 1]
```diff
@@ -112,9 +164,131 @@ func waitForPFCPAgentToStart() error {
 		case <-timeout:
 			return errors.New("timed out")
 		case <-ticker:
-			if IsConnectionOpen("127.0.0.1", "8805") {
+			if IsConnectionOpen("udp", "127.0.0.1", "8805") {
```
Comment (pudelkoM) [inline test/integration/framework.go:167]:
```suggestion
			if IsConnectionOpen(net, "127.0.0.1", "8805") {
```
?

Comment (osinstom) [reply test/integration/framework.go:167]: Good catch.

**Comment by osinstom** [comment]
retest this please

**Comment by osinstom** [comment]
ok to test

**Comment by osinstom** [comment]
retest this please

**Review by pudelkoM — APPROVED** [review]

**Pull request merged**